state/modeluser.go

```diff
@@ -193,9 +193,33 @@ func (e *UserModel) LastConnection() (time.Time, error) {
 	return lastConnDoc.LastConnection, nil
 }
 
+// allUserModels returns a list of all models with the user they belong to.
+func (st *State) allUserModels() ([]*UserModel, error) {
+	models, err := st.AllModels()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var result []*UserModel
+	for _, model := range models {
+		result = append(result, &UserModel{Model: model, User: model.Owner()})
+	}
+	return result, nil
+}
+
 // ModelsForUser returns a list of models that the user
 // is able to access.
 func (st *State) ModelsForUser(user names.UserTag) ([]*UserModel, error) {
+	// Consider the controller permissions overriding Model permission, for
+	// this case the only relevant one is superuser.
+	// The mgo query below wont work for superuser case because it needs at
+	// least one model user per model.
+	access, err := st.UserAccess(user, st.controllerTag)
+	if err == nil && access.Access == permission.SuperuserAccess {
+		return st.allUserModels()
+	}
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
Please add an explicit test that deals with what happens when you get a NotFound error.